Lib/compileall.py

```diff
@@ -220,10 +229,16 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
             if not quiet:
                 print('Compiling {!r}...'.format(fullname))
             try:
-                for opt_level, cfile in opt_cfiles.items():
+                for index, opt_level in enumerate(sorted(optimize)):
+                    cfile = opt_cfiles[opt_level]
                     ok = py_compile.compile(fullname, cfile, dfile, True,
                                             optimize=opt_level,
                                             invalidation_mode=invalidation_mode)
+                    if index > 0 and hardlink_dupes:
+                        previous_cfile = opt_cfiles[optimize[index - 1]]
```

Comment:
What if optimize contains duplicated optimization levels? Like optimize=[0,0,1]?

Reply:
Oh. I tested: it doesn't work. I pushed a first: use set() to remove duplicates.

I also fixed the test on the number of different optimization levels: test also when optimize is not an integer.